enhancements/multi-arch/heterogeneous-architecture.md

```diff
@@ -0,0 +1,373 @@
+possessed by the CVO manifestlist).
+
+In either case, oc adm release new would permit non-CVO component images to be manifest or manifestlists and pass them through directly to the
+resultant release manifest(s). If a manifestlist release payload is generated, each architecture specific release payload manifest will reference
+the same pullspecs provided in the input imagestream. Since the internal Openshift registry does not support manifestlists, the release controller
+would use quay.io.
+
+### Upgrades - Cincinnati and CVO
+
+In addition to the homogeneous payload graph that we have today, [support][OCPPLAN-7642] needs to be added for an additional graph for heterogeneous
+payloads which CVO will need to look at to make upgrade recommendations. There are a few upgrade scenarios to consider with the addition of
+heterogeneous payloads:
+
+- heterogeneous -> heterogeneous release payload upgrade
+- homogeneous -> heterogeneous release payload upgrade
+- heterogeneous -> homogeneous release payload upgrade
+
+The support for homogeneous payloads will exist so customers running disconnected homogeneous clusters do not incur the cost of mirroring the
+additional architectures if they were to use heterogeneous payloads. This means that in order for CVO to find the current release in the graph, it
+will have to know about the existing homogeneous graph as well in addition to the new heterogeneous graph. There are few options on the best way
+to make update recommendations especially in the case of upgrading from a homogeneous to a heterogeneous payload:
+
+- Have the heterogeneous graph include the homogeneous releases so CVO can make recommendations based on the current homogeneous release image.
+This is complicated by the fact that there will be multiple releases with the same version but different payloads. The trigger to update to a
+heterogeneous payload would be an option provided by the administrator through either the ClusterVersion or Infrastructure API to "move" to the
+heterogeneous graph.
+- Have both the graphs be completely independent of each other and have an option in one of the aforementioned APIs to make the cluster heterogeneous
+and CVO would request the heterogeneous graph, find the current release version and update to it.
+- Have an independent channel for heterogeneous payloads and have administrators switch channels if they want to move between homogeneous and
+heterogeneous payloads
+
+These options are under discussion and a finalized proposal will be put forth in the ticket for upgrades referenced above along with the API that would
+change to include the option to indicate a move to the heterogeneous payload. For a heterogeneous to a homogeneous payload upgrade, CVO would also need
+to sanity check that there are no workers present which do not match the architecture of the homogeneous payload. If it does, it should fail the
+upgrade.
```

Comment:
Yes, this use case was seen as atypical. I'll discuss with Duncan about it, but i think we can say that a manual upgrade can be forced but this will not be officially supported.

Reply:
added this to open questions.